osu.Game/Screens/Play/HUD/SoloGameplayLeaderboard.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Scoring;
+using osu.Game.Users;
+
+namespace osu.Game.Screens.Play.HUD
+{
+    public class SoloGameplayLeaderboard : GameplayLeaderboard
+    {
+        private readonly IUser trackingUser;
+
+        public readonly IBindableList<ScoreInfo> Scores = new BindableList<ScoreInfo>();
+
+        // hold references to ensure bindables are updated.
+        private readonly List<Bindable<long>> scoreBindables = new List<Bindable<long>>();
+
+        [Resolved]
+        private ScoreProcessor scoreProcessor { get; set; } = null!;
+
+        [Resolved]
+        private ScoreManager scoreManager { get; set; } = null!;
+
+        public SoloGameplayLeaderboard(IUser trackingUser)
+        {
+            this.trackingUser = trackingUser;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            Scores.BindCollectionChanged((_, _) => Scheduler.AddOnce(showScores), true);
+        }
+
+        private void showScores()
+        {
+            Clear();
+            scoreBindables.Clear();
+
+            if (!Scores.Any())
+                return;
+
+            ILeaderboardScore local = Add(trackingUser, true);
+
+            local.TotalScore.BindTarget = scoreProcessor.TotalScore;
+            local.Accuracy.BindTarget = scoreProcessor.Accuracy;
+            local.Combo.BindTarget = scoreProcessor.Combo;
+
+            // Local score should always show lower than any existing scores in cases of ties.
+            local.DisplayOrder.Value = long.MaxValue;
+
+            foreach (var s in Scores)
```

Comment:
Once player loads, the local player is initially placed at the top of the leaderboard then sorted back down a moment later. I think adding it after existing scores would work better.

![CleanShot 2022-09-20 at 23 39 13@2x](https://user-images.githubusercontent.com/22781491/191360327-53d0d901-09cc-4000-8adb-64f610832b46.png)

Reply:
I actually thought this was an issue with the base leaderboard implementation and was going to leave for a follow-up fix, but this makes sense.